pyro/distributions/empirical.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import absolute_import, division, print_function
+
+import numbers
+
+import math
+import torch
+
+from pyro.distributions.torch_distribution import TorchDistribution
+from pyro.distributions.torch import Categorical
+from pyro.distributions.util import copy_docs_from, is_validation_enabled
+from pyro.distributions.util import log_sum_exp
+
+
+@copy_docs_from(TorchDistribution)
+class Empirical(TorchDistribution):
+    r"""
+    Empirical distribution associated with the sampled data.
+    """
+
+    arg_constraints = {}
+
+    def __init__(self):
```

Comment:
Can  you pass through the `validate_args` arg as other `TorchDistributions` do? Then you can use the idomatic `if self._validate_args` rather than `if is_validation_enabled()`?

Reply:
Good idea, will change.